frontend/src/metabase/actions/containers/ActionCreator/utils.unit.spec.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  createMockActionFormSettings,
+  createMockFieldSettings,
+  createMockParameter,
+} from "metabase-types/api/mocks";
+import { syncFieldsWithParameters } from "./utils";
+
+describe("syncFieldsWithParameters", () => {
+  it("should not modify settings if there are no changed parameters", () => {
+    const settings = createMockActionFormSettings({
+      fields: {
+        u1: createMockFieldSettings({ id: "u1" }),
+        u2: createMockFieldSettings({ id: "u2" }),
+      },
+    });
+    const parameters = [
+      createMockParameter({
+        id: "u1",
+      }),
+      createMockParameter({
+        id: "u2",
+      }),
+    ];
+
+    const newSettings = syncFieldsWithParameters(settings, parameters);
+
+    expect(newSettings).toBe(settings);
+  });
+
+  it("should add new fields and remove non-existing ones", () => {
+    const settings = createMockActionFormSettings({
+      fields: {
+        u1: createMockFieldSettings({ id: "u1" }),
+        u2: createMockFieldSettings({ id: "u2", fieldType: "number" }),
+      },
+    });
+    const parameters = [
+      createMockParameter({
+        id: "u2",
+      }),
+      createMockParameter({
+        id: "u3",
+      }),
+    ];
+
+    const newSettings = syncFieldsWithParameters(settings, parameters);
+
+    expect(newSettings).toMatchObject({
+      fields: {
+        u2: { id: "u2", fieldType: "number" },
+        u3: { id: "u3", fieldType: "string" },
+      },
+    });
+    expect(newSettings).not.toMatchObject({
+      u1: { id: "u1" },
```

Comment:
are you supposed to nest `u1` in `fields` like the assertion above? I assume you need 2 assertions because `toMatchObject` partially matches the result. What do you think about using `.toEqual` to assert the field IDs here instead (in addition to the existing `.toMatchObject`)? something like
```ts
expect(Object.keys(newSettings.fields)).toEqual(["u2", "u3"])
```